overlord/snapstate/snapstate.go

```diff
@@ -110,6 +110,20 @@ func doInstall(s *state.State, curActive bool, ss *SnapSetup) (*state.TaskSet, e
 	addTask(linkSnap)
 	linkSnap.WaitFor(setupSecurity)
 
+	// do GC!
+	if !snapst.Current.Unset() {
+		prev := linkSnap
+		seq := snapst.Sequence
+		currentIndex := snapst.findIndex(snapst.Current)
+		for i := currentIndex - 2; i >= 0; i-- {
```

Comment:
It'd be very slightly nicer to see revisions moving up in the removals. 0 => currentIndex-2